src/Snackbar/Snackbar.types.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  SnackbarProps as MuiSnackbarProps,
+  SnackbarContentProps
+} from "@mui/material";
+
+export interface SnackbarProps {
+  /**
+   * Callback fired when the user clicks the action button.
+   *
+   * **Signature**
+   * ```
+   * function(event: object) => void
+   * ```
+   * _event_: The event source of the callback.
+   */
+  actionCallback?: (event: React.SyntheticEvent<HTMLElement>) => void;
+  /**
+   * The action button text to display. It renders after the message, at the end of the snackbar.
+   */
+  actionText?: string;
+  /**
+   * The number of milliseconds to wait before automatically calling the onClose function. onClose should then set the state of the open prop to hide the Snackbar. This behavior is disabled by default with the null value.
+   */
+  autoHideDuration?: MuiSnackbarProps["autoHideDuration"];
+  /**
+   * The message to display.
+   */
+  message: SnackbarContentProps["message"];
+  /**
+   * Callback fired when the component requests to be closed. Typically onClose is used to set state in the parent component, which is used to control the Snackbar open prop.
+   *
+   * **Signature**
+   * ```
+   * function(event, reason) => void
+   * ```
+   *
+   * _event_: The event source of the callback.
+   *
+   * _reason_: Can be: "timeout" (autoHideDuration expired), "clickaway", "action".
+   */
+  onClose: MuiSnackbarProps["onClose"];
```

Comment:
Previously, onClose was an optional prop, but now it's a required prop was it an intentional change? I am asking this because we have autoHideDuration in such cases we don't want onClose to be a required prop right? please correct me if I am wrong,

Reply:
This wasn't intentional no so good spot. MUI itself has this optional I think so we should probably follow suit. However, I think your understanding of autoHideDuration is not quite right. When autoHideDuration is set it just automatically calls onClose function after the given number of milliseconds. It doesn't actually close the dialog, that is the job of the onClose and the parent component which is controlling the open state. See the help comments for autoHideDuration prop: "The number of milliseconds to wait before automatically calling the onClose function. onClose should then set the state of the open prop to hide the Snackbar. This behavior is disabled by default with the null value." I don't actually see it possible to use the component in a controlled manner without using open and onClose props together but we can keep onClose optional to follow MUI and ensure nothing breaks.